statefun-e2e-tests/statefun-smoke-e2e/src/main/java/org/apache/flink/statefun/e2e/smoke/Fn.java

```diff
@@ -17,23 +17,36 @@
  */
 package org.apache.flink.statefun.e2e.smoke;
 
+import org.apache.flink.statefun.sdk.java.Context;
+import org.apache.flink.statefun.sdk.java.StatefulFunction;
+import org.apache.flink.statefun.sdk.java.TypeName;
+import org.apache.flink.statefun.sdk.java.ValueSpec;
+import org.apache.flink.statefun.sdk.java.StatefulFunctionSpec;
+import org.apache.flink.statefun.sdk.java.message.Message;
+
 import java.util.Objects;
-import org.apache.flink.statefun.sdk.Context;
-import org.apache.flink.statefun.sdk.StatefulFunction;
-import org.apache.flink.statefun.sdk.annotations.Persisted;
-import org.apache.flink.statefun.sdk.state.PersistedValue;
+import java.util.concurrent.CompletableFuture;
 
 public class Fn implements StatefulFunction {
 
-  @Persisted private final PersistedValue<Long> state = PersistedValue.of("state", Long.class);
-  private final CommandInterpreter interpreter;
+  static final TypeName TYPENAME = TypeName.typeNameOf("statefun.e2e", "interpreter");
+  static final ValueSpec<Long> state = ValueSpec.named("state").withLongType();
+
+  final StatefulFunctionSpec SPEC;
```

Comment:
I believe this should be static, as you will need the spec to register the function.